SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -146,15 +145,37 @@ bool USpatialActorChannel::CleanUp(const bool bForDestroy, EChannelCloseReason C
 	}
 #endif
 
-	// Must cleanup actor and subobjects before UActorChannel::Cleanup as it will clear CreateSubObjects
-	Receiver->CleanupDeletedEntity(EntityId);
+	if (CloseReason == EChannelCloseReason::Dormancy)
+	{
+		NetDriver->RegisterDormantEntityId(EntityId);
+	}
+	else
+	{
+		// Must cleanup actor and subobjects before UActorChannel::Cleanup as it will clear CreateSubObjects
+		NetDriver->PackageMap->RemoveEntityActor(EntityId);
+	}
+
+	NetDriver->RemoveActorChannel(EntityId);
 
 	return UActorChannel::CleanUp(bForDestroy, CloseReason);
 }
 
 int64 USpatialActorChannel::Close(EChannelCloseReason Reason)
 {
-	DeleteEntityIfAuthoritative();
+	if (Reason != EChannelCloseReason::Dormancy)
+	{
+		DeleteEntityIfAuthoritative();
```

Comment:
I think we also need to remove the entity from the package map here.

Reply:
I think you are correct